lib/modules/platform/scm-manager/index.ts

```diff
@@ -0,0 +1,333 @@
+
+export async function getBranchPr(branchName: string): Promise<Pr | null> {
+  return await findPr({ branchName, state: 'open' });
+}
+
+export async function findPr({
+  branchName,
+  prTitle,
+  state = 'all',
+}: FindPRConfig): Promise<Pr | null> {
+  const inProgressPrs = await getPrList();
+  const result = inProgressPrs.find(
+    (pr) =>
+      branchName === pr.sourceBranch &&
+      (!prTitle || prTitle === pr.title) &&
+      matchPrState(pr, state),
+  );
+
+  if (result) {
+    logger.info(`Found PR ${JSON.stringify(result)}`);
+    return result;
+  }
+
+  logger.debug(
+    `Could not find PR with source branch ${branchName} and title ${
+      prTitle ?? ''
+    } and state ${state}`,
+  );
+
+  return null;
+}
+
+export async function getPr(number: number): Promise<Pr | null> {
+  const inProgressPrs = await getPrList();
+  const cachedPr = inProgressPrs.find((pr) => pr.number === number);
+
+  if (cachedPr) {
+    logger.info(`Returning from cached PRs, ${JSON.stringify(cachedPr)}`);
+    return cachedPr;
+  }
+
+  try {
+    const result = await scmmClient.getRepoPr(config.repository, number);
+    logger.info(`Returning PR from API, ${JSON.stringify(result)}`);
+    return mapPrFromScmToRenovate(result);
+  } catch (error) {
+    logger.info(`Not found PR with id ${number}`);
+    return null;
+  }
+}
+
+export async function getPrList(): Promise<Pr[]> {
+  if (config.prList === null) {
+    try {
+      config.prList = (await scmmClient.getAllRepoPrs(config.repository)).map(
+        (pr) => mapPrFromScmToRenovate(pr),
+      );
+    } catch (error) {
+      logger.error(error);
+    }
+  }
+
+  return config.prList ?? [];
+}
+
+export async function createPr({
+  sourceBranch,
+  targetBranch,
+  prTitle,
+  prBody,
+  draftPR,
+}: CreatePRConfig): Promise<Pr> {
+  const createdPr = await scmmClient.createPr(config.repository, {
+    source: sourceBranch,
+    target: targetBranch,
+    title: prTitle,
+    description: sanitize(prBody),
+    status: draftPR ? 'DRAFT' : 'OPEN',
+  });
+
+  logger.info(
+    `PR created with title '${createdPr.title}' from source '${createdPr.source}' to target '${createdPr.target}'`,
+  );
+  logger.debug(`PR created ${JSON.stringify(createdPr)}`);
+
+  return mapPrFromScmToRenovate(createdPr);
+}
+
+export async function updatePr({
+  number,
+  prTitle,
+  prBody,
+  state,
+  targetBranch,
+}: UpdatePrConfig): Promise<void> {
+  await scmmClient.updatePr(config.repository, number, {
+    title: prTitle,
+    description: sanitize(prBody) ?? undefined,
+    target: targetBranch,
+    status: mapPrState(state),
+  });
+
+  logger.info(`Updated PR #${number} with title ${prTitle}`);
+}
+
+export function mergePr(config: MergePRConfig): Promise<boolean> {
+  logger.debug('NO-OP mergePr');
+  return Promise.resolve(false);
+}
+
+export function getBranchStatus(
+  branchName: string,
+  internalChecksAsSuccess: boolean,
+): Promise<BranchStatus> {
+  logger.debug('NO-OP getBranchStatus');
```

Comment:
I see we still have some `NO-OP` jargon in the debug messages, I'll let the maintainers decide if this needs to be changed or not.